### PR TITLE
Unpacker configurable buffer size

### PR DESCRIPTION
**Pull request by mklander**
Config option for unpacker.buffer.size

### CODE REVIEW

File: terraform/stack/main.tf [hunk 1]
```diff
@@ -21,6 +21,7 @@ module "bag_unpacker" {
     queue_url               = "${module.bag_unpacker_queue.url}"
     queue_parallelism       = "10"
     destination_bucket_name = "${var.ingest_bucket_name}"
+    unpacker_buffer_size    = "8192"
```
Comment (alexwlchan) [inline terraform/stack/main.tf]: env_vars_length

**Comment by alexwlchan** [comment]
Hopefully this should fix https://github.com/wellcometrust/platform/issues/3481

File: bag_unpacker/src/main/scala/uk/ac/wellcome/platform/archive/bagunpacker/services/Unpacker.scala [hunk 2]
```diff
@@ -6,21 +6,26 @@ import java.time.Instant
...
 import scala.util.{Failure, Success}
 
-class Unpacker(implicit s3Client: AmazonS3, ec: ExecutionContext) {
+class Unpacker(config: UnpackerConfig)(implicit s3Client: AmazonS3,
+                                       ec: ExecutionContext) {
 
-  private val s3Uploader = new S3Uploader()
+  // See comments inside both of these classes -- it's important that they
+  // both have the same buffer size.
+  private val s3Uploader = new S3Uploader(bufferSize = config.bufferSize)
```
Comment (mklander) [inline bag_unpacker/src/main/scala/uk/ac/wellcome/platform/archive/bagunpacker/services/Unpacker.scala]: from the SDK best practice docs should this be `bufferSize+1`?

Comment (alexwlchan) [reply bag_unpacker/src/main/scala/uk/ac/wellcome/platform/archive/bagunpacker/services/Unpacker.scala]: Sure

**Pull request merged**